hack/clean/clean.go

```diff
@@ -0,0 +1,96 @@
+package main
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"flag"
+	"os"
+	"strings"
+	"time"
+
+	mgmtfeatures "github.com/Azure/azure-sdk-for-go/services/resources/mgmt/2019-07-01/features"
+	"github.com/sirupsen/logrus"
+
+	utillog "github.com/Azure/ARO-RP/pkg/util/log"
+	"github.com/Azure/ARO-RP/pkg/util/purge"
+)
+
+const (
+	defaultTTL     = 48
+	defaultKeepTag = "keepit"
+)
+
+var dryRun = flag.Bool("dryRun", false, "should the purge perform dry for test")
+
+func main() {
+	ctx := context.Background()
+	log := utillog.GetLogger()
+
+	flag.Parse()
+
+	if err := run(ctx, log); err != nil {
+		log.Fatal(err)
+	}
+}
+
+func run(ctx context.Context, log *logrus.Entry) error {
+	var ttl time.Duration
+	if os.Getenv("AZURE_PURGE_TTL") != "" {
+		var err error
+		ttl, err = time.ParseDuration(os.Getenv("AZURE_PURGE_TTL"))
+		if err != nil {
+			return err
+		}
+	} else {
+		ttl = defaultTTL
+	}
+
+	deleteCheck := func(resourceGroup mgmtfeatures.ResourceGroup, log *logrus.Entry) bool {
+		if strings.HasPrefix(*resourceGroup.Name, "v4-e2e-rg-") ||
+			strings.HasPrefix(*resourceGroup.Name, "aro-v4-e2e-rg-") {
+			log.Infof("Group %s have the SAFE prefix. SKIP", *resourceGroup.Name)
+			return false
+		}
```

Comment:
I'm not understanding this logic

Reply:
This skips everything that is not prefixed with "v4-e2e-rg-" or "aro-v4-e2e-rg-".

Judging by your surprise, I am thinking are there different requirements for e2e pipeline and dev pipeline on resource group deletion?

Reply:
So this rule is applicable to MSFT subscription only. I think we might want to have this configurable as regexp as input? So we can configure it for different subscriptions? As tags will not be an option in MSFT subscription too.